cupyx/jit/_compile.py

```diff
@@ -54,6 +55,53 @@ def new_func(node, *args, **kwargs):
     return new_func
 
 
+def _parse_function_object(func):
+    # Parses function object into ast.FunctionDef object.
+    if not callable(func):
+        raise ValueError('`func` must be a callable object.')
+
+    if jit._getsource_func is not None:
+        full_source = jit._getsource_func(func)
+        start_line, end_line = 0, math.inf
+        source = full_source
+    else:
+        filename = inspect.getsourcefile(func)
```

Comment:
Needs error handling:
* TypeError may be raised here.
* filename can be None if func is C extension function.

Reply:
Do you mean that we need to fallback to inspect.getsource?

Reply:
No, I meant that proper error handling is needed here. 😄 
```suggestion
        try:
            filename = inspect.getsourcefile(func)
        except TypeError:
            filename = None
        if filename is None:
            raise ValueError(f'JIT needs access to Python source for {func}'
                             'but could not be located')
```